tests/Tests/ORM/Functional/Ticket/GH10808Test.php

```diff
@@ -55,14 +52,28 @@ public function testDQLDeferredEagerLoad(): void
 
         $eagerLoadResult = $query->setHint(UnitOfWork::HINT_DEFEREAGERLOAD, false)->getSingleResult();
 
-        self::assertNotEquals(
-            GH10808AppointmentChild::class,
-            get_class($deferredLoadResult->child),
-            '$deferredLoadResult->child should be a proxy',
+        if (! $this->_em->getConfiguration()->isNativeLazyObjectsEnabled()) {
+            self::assertNotEquals(
+                GH10808AppointmentChild::class,
+                get_class($deferredLoadResult->child),
+                '$deferredLoadResult->child should be a proxy',
+            );
+            self::assertEquals(
+                GH10808AppointmentChild::class,
+                get_class($eagerLoadResult->child),
+                '$eagerLoadResult->child should not be a proxy',
+            );
+
+            return;
+        }
+
+        $reflector = new ReflectionClass(GH10808AppointmentChild::class);
+        self::assertFalse(
+            $reflector->isUninitializedLazyObject($deferredLoadResult->child),
```

Comment:
use `$this->isUninitializedObject(...)` here, which allows using the same code for any config of the proxies (that's what we use in other tests)